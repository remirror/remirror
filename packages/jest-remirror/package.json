{
  "name": "jest-remirror",
  "version": "3.0.0-beta.6",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/jest-remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/jest-remirror.js",
      "require": "./dist/jest-remirror.cjs"
    },
    "./environment": {
      "types": "./dist-types/environment.d.ts",
      "import": "./environment/dist/jest-remirror-environment.js",
      "require": "./environment/dist/jest-remirror-environment.cjs"
    },
    "./package.json": "./package.json"
  },
  "main": "./dist/jest-remirror.cjs",
  "module": "./dist/jest-remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "environment"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.6",
    "@remirror/pm": "3.0.0-beta.4",
    "@remirror/preset-core": "3.0.0-beta.6",
    "@testing-library/dom": "^9.3.0",
    "@types/sanitize-html": "^2.9.0",
    "jest-prosemirror": "3.0.0-beta.4",
    "sanitize-html": "^2.10.0",
    "tiny-warning": "^1.0.3"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.0",
    "@types/node": "^16.18.34",
    "jest": "^29.5.0"
  },
  "peerDependencies": {
    "@types/node": "*",
    "jest": ">= 24.9.0"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    },
    "jest": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/jest-remirror.d.ts",
        "import": "./dist/jest-remirror.js",
        "require": "./dist/jest-remirror.cjs"
      },
      "./environment": {
        "types": "./environment/dist/jest-remirror-environment.d.ts",
        "import": "./environment/dist/jest-remirror-environment.js",
        "require": "./environment/dist/jest-remirror-environment.cjs"
      },
      "./package.json": "./package.json"
    },
    "main": "./dist/jest-remirror.cjs",
    "module": "./dist/jest-remirror.js",
    "type": "module",
    "types": "./dist/jest-remirror.d.ts"
  },
  "@remirror": {
    "tsconfigs": {
      "src": {
        "compilerOptions": {
          "types": [
            "node",
            "jest"
          ]
        }
      }
    }
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "environment.ts"
    ]
  }
}
