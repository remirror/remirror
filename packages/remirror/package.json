{
  "name": "remirror",
  "version": "3.0.0-beta.8",
  "description": "One editing package to rule them all, one editing package to bind them.",
  "keywords": [
    "prosemirror",
    "react",
    "remirror",
    "wysiwyg",
    "editing",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>",
    "ocavue "
  ],
  "sideEffects": false,
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/remirror.js",
      "require": "./dist/remirror.cjs"
    },
    "./dom": {
      "types": "./dist-types/dom.d.ts",
      "import": "./dom/dist/remirror-dom.js",
      "require": "./dom/dist/remirror-dom.cjs"
    },
    "./extensions": {
      "types": "./dist-types/extensions.d.ts",
      "import": "./extensions/dist/remirror-extensions.js",
      "require": "./extensions/dist/remirror-extensions.cjs"
    },
    "./icons": {
      "types": "./dist-types/icons.d.ts",
      "import": "./icons/dist/remirror-icons.js",
      "require": "./icons/dist/remirror-icons.cjs"
    },
    "./package.json": "./package.json",
    "./styles/*": "./styles/*"
  },
  "main": "./dist/remirror.cjs",
  "module": "./dist/remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "dom",
    "extensions",
    "icons",
    "styles"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.8",
    "@remirror/core-constants": "3.0.0-beta.4",
    "@remirror/core-helpers": "4.0.0-beta.5",
    "@remirror/core-types": "3.0.0-beta.5",
    "@remirror/core-utils": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.8",
    "@remirror/extension-annotation": "3.0.0-beta.8",
    "@remirror/extension-bidi": "3.0.0-beta.8",
    "@remirror/extension-blockquote": "3.0.0-beta.8",
    "@remirror/extension-bold": "3.0.0-beta.8",
    "@remirror/extension-callout": "3.0.0-beta.8",
    "@remirror/extension-code": "3.0.0-beta.8",
    "@remirror/extension-code-block": "3.0.0-beta.8",
    "@remirror/extension-collaboration": "3.0.0-beta.8",
    "@remirror/extension-columns": "3.0.0-beta.8",
    "@remirror/extension-diff": "3.0.0-beta.8",
    "@remirror/extension-doc": "3.0.0-beta.8",
    "@remirror/extension-drop-cursor": "3.0.0-beta.8",
    "@remirror/extension-embed": "3.0.0-beta.8",
    "@remirror/extension-emoji": "3.0.0-beta.8",
    "@remirror/extension-entity-reference": "3.0.0-beta.8",
    "@remirror/extension-epic-mode": "3.0.0-beta.8",
    "@remirror/extension-events": "3.0.0-beta.8",
    "@remirror/extension-find": "1.0.0-beta.8",
    "@remirror/extension-font-family": "3.0.0-beta.8",
    "@remirror/extension-font-size": "3.0.0-beta.8",
    "@remirror/extension-gap-cursor": "3.0.0-beta.8",
    "@remirror/extension-hard-break": "3.0.0-beta.8",
    "@remirror/extension-heading": "3.0.0-beta.8",
    "@remirror/extension-history": "3.0.0-beta.8",
    "@remirror/extension-horizontal-rule": "3.0.0-beta.8",
    "@remirror/extension-image": "3.0.0-beta.8",
    "@remirror/extension-italic": "3.0.0-beta.8",
    "@remirror/extension-link": "3.0.0-beta.8",
    "@remirror/extension-list": "3.0.0-beta.8",
    "@remirror/extension-markdown": "3.0.0-beta.8",
    "@remirror/extension-mention": "3.0.0-beta.8",
    "@remirror/extension-mention-atom": "3.0.0-beta.8",
    "@remirror/extension-node-formatting": "3.0.0-beta.8",
    "@remirror/extension-paragraph": "3.0.0-beta.8",
    "@remirror/extension-placeholder": "3.0.0-beta.8",
    "@remirror/extension-positioner": "3.0.0-beta.8",
    "@remirror/extension-shortcuts": "3.0.0-beta.8",
    "@remirror/extension-strike": "3.0.0-beta.8",
    "@remirror/extension-sub": "3.0.0-beta.8",
    "@remirror/extension-sup": "3.0.0-beta.8",
    "@remirror/extension-tables": "3.0.0-beta.8",
    "@remirror/extension-text": "3.0.0-beta.8",
    "@remirror/extension-text-case": "3.0.0-beta.8",
    "@remirror/extension-text-color": "3.0.0-beta.8",
    "@remirror/extension-text-highlight": "3.0.0-beta.8",
    "@remirror/extension-trailing-node": "3.0.0-beta.8",
    "@remirror/extension-underline": "3.0.0-beta.8",
    "@remirror/extension-whitespace": "3.0.0-beta.8",
    "@remirror/icons": "3.0.0-beta.5",
    "@remirror/preset-core": "3.0.0-beta.8",
    "@remirror/preset-formatting": "3.0.0-beta.8",
    "@remirror/preset-wysiwyg": "3.0.0-beta.8",
    "@remirror/theme": "3.0.0-beta.5",
    "@types/refractor": "^3.0.2",
    "refractor": "^3.6.0"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.1",
    "@remirror/pm": "3.0.0-beta.6",
    "@types/prettier": "^2.7.2",
    "prettier": "^2.8.8"
  },
  "peerDependencies": {
    "@remirror/pm": "^3.0.0-beta.6",
    "@types/prettier": "^2.7.2",
    "prettier": "^2.8.8"
  },
  "peerDependenciesMeta": {
    "@types/prettier": {
      "optional": true
    },
    "prettier": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/remirror.d.ts",
        "import": "./dist/remirror.js",
        "require": "./dist/remirror.cjs"
      },
      "./dom": {
        "types": "./dom/dist/remirror-dom.d.ts",
        "import": "./dom/dist/remirror-dom.js",
        "require": "./dom/dist/remirror-dom.cjs"
      },
      "./extensions": {
        "types": "./extensions/dist/remirror-extensions.d.ts",
        "import": "./extensions/dist/remirror-extensions.js",
        "require": "./extensions/dist/remirror-extensions.cjs"
      },
      "./icons": {
        "types": "./icons/dist/remirror-icons.d.ts",
        "import": "./icons/dist/remirror-icons.js",
        "require": "./icons/dist/remirror-icons.cjs"
      },
      "./package.json": "./package.json",
      "./styles/*": "./styles/*"
    },
    "main": "./dist/remirror.cjs",
    "module": "./dist/remirror.js",
    "type": "module",
    "types": "./dist/remirror.d.ts"
  },
  "@remirror": {},
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "dom.ts",
      "extensions.ts",
      "icons.ts"
    ]
  }
}
