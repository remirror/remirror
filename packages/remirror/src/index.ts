export * from '@remirror/core';
export * from '@remirror/theme';
