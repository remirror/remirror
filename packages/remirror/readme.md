# remirror

> One package to rule them all, one entry point to bind them.

[![Version][version]][npm] [![Weekly Downloads][downloads-badge]][npm] [![Bundled size][size-badge]][size] [![Typed Codebase][typescript]](#) [![MIT License][license]](#)

[version]: https://flat.badgen.net/npm/v/remirror
[npm]: https://npmjs.com/package/remirror
[license]: https://flat.badgen.net/badge/license/MIT/purple
[size]: https://bundlephobia.com/result?p=remirror
[size-badge]: https://flat.badgen.net/bundlephobia/minzip/remirror
[typescript]: https://flat.badgen.net/badge/icon/TypeScript?icon=typescript&label
[downloads-badge]: https://badgen.net/npm/dw/remirror/red?icon=npm

## Installation

```bash
# yarn
yarn add remirror @remirror/pm

# pnpm
pnpm add remirror @remirror/pm

# npm
npm install remirror @remirror/pm
```

The `remirror` package will automatically install the `@remirror/pm` package for you. You only need to install it yourself if you'd like to ensure consistent versions of the `prosemirror-*` libraries when importing from `@remirror/pm/state` instead of `prosemirror-state` or `@remirror/pm/model` instead of `prosemirror-model`.

## Usage

Rather than installing multiple scoped packages, the `remirror` package is a gateway to using all the goodness that remirror provides while minimising your bundle size.

The following creates a controlled editor with React.

```tsx
import React from 'react';
import { SocialPreset } from 'remirror/extensions';
import { Remirror, SocialEmojiComponent, useRemirror } from '@remirror/react';

const EditorWrapper = () => {
  const socialPreset = new SocialPreset();
  const { state, onChange } = useRemirror({ extensions: () => [...socialPreset()] });

  return (
    <Remirror state={state} onChange={onChange} manager={manager} autoRender={true}>
      <SocialEmojiComponent />
    </Remirror>
  );
};
```

These are the entry points available through the `remirror` package.

- `remirror` - All the core functionality available through `@remirror/core`.
- `remirror/extensions` - All the core extensions and presets made available through the main `remirror` repository. This doesn't include any framework specific extensions and presets.
- `remirror/dom` - The dom framework implementation of via `createDomEditor`.
