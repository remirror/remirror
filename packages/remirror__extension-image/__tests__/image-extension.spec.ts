import { jest } from '@jest/globals';
import { extensionValidityTest, renderEditor } from 'jest-remirror';
import { DelayedPromiseCreator } from 'remirror';
import { delay } from 'testing';

import { ImageAttributes, ImageExtension } from '../';

extensionValidityTest(ImageExtension);

describe('commands', () => {
  describe('insertImage', () => {
    it('can insert images', () => {
      const editor = renderEditor(() => [new ImageExtension()]);
      const { doc, p } = editor.nodes;
      const { image } = editor.attributeNodes;

      editor.add(doc(p('content <cursor>')));
      editor.commands.insertImage({ src: 'https://test.com' });

      expect(editor.doc).toEqualProsemirrorNode(
        doc(p('content ', image({ src: 'https://test.com' })())),
      );
    });
  });

  describe('uploadImage', () => {
    jest.useFakeTimers();

    afterAll(() => {
      jest.useRealTimers();
    });

    it('can upload images', async () => {
      const promise: Promise<ImageAttributes> = delay(100).then(() => ({
        src: 'https://test.com',
      }));
      const editor = renderEditor(() => [new ImageExtension()]);
      const { doc, p } = editor.nodes;
      const { image } = editor.attributeNodes;

      editor.add(doc(p('content <cursor>')));
      const delayedImage: DelayedPromiseCreator<ImageAttributes> = () => promise;

      editor.commands.uploadImage(delayedImage);

      expect(editor.dom).toMatchSnapshot();

      jest.runAllTimers();
      await promise;

      expect(editor.doc).toEqualProsemirrorNode(
        doc(p('content ', image({ src: 'https://test.com' })())),
      );
    });
  });
});
