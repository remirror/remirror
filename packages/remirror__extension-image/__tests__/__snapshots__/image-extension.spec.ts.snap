// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands uploadImage can upload images 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    content 
    <div
      class="remirror-image-loader placeholder ProseMirror-widget"
    />
    <img
      alt=""
      class="ProseMirror-separator"
    />
    <br
      class="ProseMirror-trailingBreak"
    />
  </p>
</div>
`;
